### PR TITLE
chef vault mysql check and galera password

### PR DESCRIPTION
**Pull request by DrRayWang**
Chef vault mysql check password and mysql galera password, and made related changes in recipes and template files

### CODE REVIEW

**Comment by cbaenziger** [comment]
This will need a fix for Chef-BACH #244

**Comment by cbaenziger** [comment]
Please also secure the [`/etc/mysql/conf.d/wsrep.cnf`](https://github.com/bloomberg/chef-bach/blob/4fbbccced71628f9cce4bfd2e3e9ef9d0f5c5675/cookbooks/bcpc/recipes/mysql.rb#L77-L83) permissions to be more restrictive than 644.

File: cookbooks/bcpc/recipes/mysql.rb [hunk 1]
```diff
@@ -40,9 +73,9 @@
 bash "initial-mysql-config" do
   code <<-EOH
         mysql -u root -e "DROP USER ''@'localhost';
-                          GRANT USAGE ON *.* to '#{get_config('mysql-galera-user')}'@'%' IDENTIFIED BY '#{get_config('mysql-galera-password')}';
-                          GRANT ALL PRIVILEGES on *.* TO '#{get_config('mysql-galera-user')}'@'%' IDENTIFIED BY '#{get_config('mysql-galera-password')}';
-                          GRANT PROCESS ON *.* to '#{get_config('mysql-check-user')}'@'localhost' IDENTIFIED BY '#{get_config('mysql-check-password')}';
+                          GRANT USAGE ON *.* to '#{get_config('mysql-galera-user')}'@'%' IDENTIFIED BY '#{get_config('password','mysql-galera','os')}';
+                          GRANT ALL PRIVILEGES on *.* TO '#{get_config('mysql-galera-user')}'@'%' IDENTIFIED BY '#{get_config('password','mysql-galera','os')}';
+                          GRANT PROCESS ON *.* to '#{get_config('mysql-check-user')}'@'localhost' IDENTIFIED BY '#{get_config('password','mysql-check','os')}';
```
Comment (cbaenziger) [inline cookbooks/bcpc/recipes/mysql.rb]: We should use the [`sensitive`](https://docs.chef.io/resource_execute.html) attribute as appropriate to obscure the password containing commands.

Comment (DrRayWang) [inline cookbooks/bcpc/recipes/mysql.rb]: Bash resource does not have sensitive attribute

**Comment by DrRayWang** [comment]
his pull request is covered by PR#260, so can be merged in PR#260.

**Comment by amithkanand** [comment]
As discussed with Ray, I am closing this one.

**Pull request closed without merging**